### PR TITLE
etcdserver: Move all get/put/delete on AuthUsers and AuthRoles to buckets module

### PR DESCRIPTION
**Pull request by serathius**
cc @ptabor 

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/mvcc/buckets/auth_roles.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 The etcd Authors
```
Comment (ptabor) [inline server/mvcc/buckets/auth_roles.go:1]: 2021 ;)

File: server/mvcc/buckets/auth_users.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 The etcd Authors
```
Comment (ptabor) [inline server/mvcc/buckets/auth_users.go:1]: 2021 ;)

**Review by ptabor — APPROVED** [review]

File: server/mvcc/buckets/auth_users.go (hunk 2 above)
Comment (serathius) [reply server/mvcc/buckets/auth_users.go]: Fixed.

File: server/mvcc/buckets/auth_roles.go (hunk 1 above)
Comment (serathius) [reply server/mvcc/buckets/auth_roles.go]: Fixed

**Pull request merged**